while-loops/problem-set-while-loops.md

```diff
@@ -0,0 +1,849 @@
+
+Provide an implementation for `silly_sum` so that it
+- reads numbers from the user (use `input_int`) summing as we go until either
+    - the user enters `0`, or
+    - the sum reaches or exceeds `1000`
+
+##### !end-question
+
+##### !placeholder
+
+```python
+def silly_sum():
+    # Be sure to call input_int when you need an integer from the user.
+    # A default implementation is provided above for experimenting in VS Code or repl.it.
+    # You should not paste that implementation here. It will be available automatically.
+    pass
+```
+
+##### !end-placeholder
+
+##### !tests
+
+```python
+import unittest
+import main
+from main import silly_sum
+
+num_src = None
+
+def input_ints(nums):
+    for num in nums:
+        yield num
+
+def input_int(*args):
+    return next(num_src)
+
+# inject into main
+main.input_int = input_int
+
+class TestPython1(unittest.TestCase):
+
+    def test_one(self):
+        global num_src
+        num_src = input_ints([1, 1, 2, 3, 5, 8, 0, 9])
+        self.assertEqual(20, silly_sum(), 'the provided input was 1, 1, 2, 3, 5, 8, 0, 9')
+
+    def test_two(self):
+        global num_src
+        num_src = input_ints([400, 400, 400, 400])
+        self.assertEqual(1200, silly_sum(), 'the provided input was 400, 400, 400, 400')
+
+    def test_three(self):
+        global num_src
+        num_src = input_ints([999, 1, 1])
+        self.assertEqual(1000, silly_sum(), 'the provided input was 999, 1, 1')
+```
+
+##### !end-tests
+
+#### !hint
+
+A possible interaction with a user could look like this:
+> Enter an integer (0 to stop): 999<br />
+> Enter an integer (0 to stop): 1<br />
+> The sum is: 1000
+
+Here's another possible example:
+> Enter an integer (0 to stop): 400<br />
+> Enter an integer (0 to stop): 400<br />
+> Enter an integer (0 to stop): 400<br />
+> The sum is: 1200
+
+This sum is halted by the user:
+> Enter an integer (0 to stop): 1<br />
+> Enter an integer (0 to stop): 2<br />
+> Enter an integer (0 to stop): 3<br />
+> Enter an integer (0 to stop): 0<br />
+> The sum is: 6
+
+Note that printing the final sum line is not a listed requirement, but feel free to add additional output to your implementation to help think through the problem. It will not affect the results of the tests. Even if you _do_ print out the final sum, be sure to also _return_ it!
+
+#### !end-hint
+
+#### !explanation
+
+An example of a working implementation:
+
+```python
+def silly_sum():
+    # accumulate the sum
+    my_sum = 0
+
+    # stores the most recent user input
+    # default to anything except 0 (think about why this is)
+    num = 1
+
+    # check our 2 conditions:
+    #   user input must not be 0
+    #   don't ask for more input once the sum reaches at least 1000
+    while num != 0 and my_sum < 1000:
+        # prompt for input
+        num = input_int('Enter an integer (0 to stop): ')
```

Comment:
From the information provided, I didn't think that I should add a string to `input_int`. If this is something we expect the students to have been able to organically write, I would change something about the intro to `input_int` or not include this arg?

Reply:
Do we not show using `input` that way anywhere?!??!!?

Reply:
I'm seeing cases where we show `input` being used with a prompt in our coursework, so I think this should be ok.  The test _doesn't_ check for any output at all (either for a prompt or for a result), so I feel okay with leaving the prompt as a nicety in the solution.